extension/js/common/api/email-provider/gmail/google-auth.ts

```diff
@@ -184,6 +161,18 @@ export class GoogleAuth {
     return authRes;
   };
 
+  public static oauthLogin = async (url: string): Promise<string> => {
+    return await new Promise((resolve, reject) => {
+      void OAuth2.launchWebAuthFlow(url, (redirectUrl: string) => {
+        if (chrome.runtime.lastError) {
+          reject(chrome.runtime.lastError.message);
+          return;
+        }
+        resolve(redirectUrl);
+      });
+    });
+  };
```

Comment:
This `if (chrome.runtime.lastError) {` is rather indirect, but I suppose you had some reason for it? As opposed to just throwing the error where it happens and then catching it normally? (or in async lingo, rejecting the promise where it happens, and then catching that rejection normally in async try/catch.)

When this error happens, exactly where does it happen, which code generates it? It would be better to catch and handle the error right there.